src/idpi/operators/pot_vortic.py

```diff
@@ -78,94 +52,25 @@ def fpotvortic(
         xr.DataArray: The potential vorticity
 
     """
-    stpt = cast(
-        Callable[[str], dict[str, slice]],
-        partial(stpt_, diff_type=diff_type) if diff_type else stpt_,
-    )
-
-    # prepare parameters
-    dlon = U.attrs["GRIB_iDirectionIncrementInDegrees"]
-    dlat = U.attrs["GRIB_jDirectionIncrementInDegrees"]
-    deg2rad = np.pi / 180
-    inv_dlon = 1 / (dlon * deg2rad)
-    inv_dlat = 1 / (dlat * deg2rad)
-
     # target coordinates
-    # the output array is missing the border
-    mask = {"x": slice(1, -1), "y": slice(1, -1)}
-    lat = P["latitude"][mask]
-
-    # FD (finite differences) weights
-    wi = 0.5 * inv_dlon
-    wj = 0.5 * inv_dlat
-    wk = 0.5 if diff_type == "center" else 1.0
-
-    zmax = U.sizes.get("generalVerticalLayer")
-
-    def prepare_array(array):
-        if "generalVertical" in array.dims:
-            return (
-                array.rename(generalVertical="z")
-                .isel({"z": slice(zmax)})
-                .drop_indexes("z")
-            )
-        elif "generalVerticalLayer" in array.dims:
-            return array.rename(generalVerticalLayer="z").drop_indexes("z")
-        return array
-
-    U, V, W, P, T, HHL, QV, QC, QI = map(
-        prepare_array,
-        (U, V, W, P, T, HHL, QV, QC, QI),
-    )
-    if QW_load:
-        QW_load = QW_load.drop_indexes("z")
-
-    # computation
-
-    # metric coefficients at the scalar grid positions
-
-    # shortcut for stencils of hhl
-    def hhl(s: str) -> xr.DataArray:
-        return HHL[stpt(s)]
-
-    sqrtg_r_s = 1.0 / (hhl("ccc") - hhl("ccp"))
-    dzeta_dlam = (
-        0.25
-        * inv_dlon
-        * sqrtg_r_s
-        * ((hhl("pcc") - hhl("mcc")) + (hhl("pcp") - hhl("mcp")))
-    )
-    dzeta_dphi = (
-        0.25
-        * inv_dlat
-        * sqrtg_r_s
-        * ((hhl("cpc") - hhl("cmc")) + (hhl("cpp") - hhl("cmp")))
-    )
-
-    # curl
-    curl1, curl2, curl3 = curl(U, V, W, HHL, inv_dlon, inv_dlat, wk, sqrtg_r_s, lat)
-
-    # potential temperature
-    theta = ftheta(P, T)
+    deg2rad = np.pi / 180
+    lat = (rho_tot["latitude"] * deg2rad).astype(np.float32)
+    rlat = get_rotated_latitude(w)
 
-    # shortcut for stencils of theta
-    def t(s: str) -> xr.DataArray:
-        return theta[stpt(s)]
+    # compute curl
+    curl1, curl2, curl3 = curl(u, v, w, rlat, total_diff)
 
     # coriolis terms
-    cor2 = 2 * const.pc_omega / const.earth_radius * np.cos(lat * deg2rad)
-    cor3 = 2 * const.pc_omega * np.sin(lat * deg2rad)
+    cor2 = 2 * const.pc_omega / const.earth_radius * np.cos(lat)
+    cor3 = 2 * const.pc_omega * np.sin(lat)
 
-    # total air density
-    rho = f_rho_tot(T, P, QV, QC, QI, QW_load)
+    dt_dlam = total_diff.d_dlam(diff.dx(theta), diff.dz(theta))
```

Comment:
question about the naming: what is the meaning of total_diff (or class TotalDiff)?

Reply:
We discussed this last week and I was asking if you had any ideas for better naming. The class provides operators that take into account the additional terms due to the terrain following grid deformation. It is however lacking the terms due to the curvature of the earth so total diff is not the most appropriate term.